keras/engine/topology.py

```diff
@@ -465,6 +465,17 @@ def __call__(self, x, mask=None):
             x: can be a tensor or list/tuple of tensors.
             mask: tensor or list/tuple of tensors.
         '''
+        # place the op on the right device:
+        if type(x) in [list, tuple] and not hasattr(x[0], '_keras_on_device'):
```

Comment:
Any way to completely abstract this away in the Theano backend? We should not adjust the backend-agnostic part of the codebase to adjust for Theano-specific issues.